crates/sui-verifier-transactional-tests/tests/entry_points/non_key_struct_generic.mvir

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// invalid as NoStore doesn't have store, so Obj doesn't have key
```

Comment:
It makes sense to me that this does not work, but it's interesting to think about it in a way that implies that the `key` ability is effectively erased here. Is this explicitly documented somewhere?

Reply:
Its how all abilities work :) Its documented in the Move docs. 

Like consider if you have `vector<T> has copy, drop, store` 

But you want to be able to use `vector` with types that don't have those abilities, like `Coin`, so if you have `vector<Coin>` it doesn't have `copy` or `drop` anymore.

Its much like `#[derive(Clone)]` in Rust

Reply:
I can't say this part of language spec is trivial to me, but what I gathered when reading it is more of an ability "inheritance" (just like in your vector example where vector does not have abilities of its own), rather than what this example shows, which is for the embedded struct to essentially "erase" `key` ability of `Obj`.

Reply:
`vector` DOES have abilities of its own, the abilities though are predicated on the type parameters having the required abilities.

Reply:
If `vector` had a type declaration as a struct it would be `struct vector<T> has copy, drop, store { ... }` there is no inheritance going on, just conditioning. 

Much like in Rust you would have something like `impl<T: Clone> Clone for Vec<T>`. `Vec` having `Clone` is predicated on `T` having `Clone`, but there is no inheritance